backend/jottacloud/jottacloud.go

```diff
@@ -988,12 +1061,29 @@ func (f *Fs) ListR(ctx context.Context, dir string, callback fs.ListRCallback) (
 		Path:       f.filePath(dir),
 		Parameters: url.Values{},
 	}
-	opts.Parameters.Set("mode", "list")
+	opts.Parameters.Set("mode", "liststream")
+	list := walk.NewListRHelper(callback)
 
 	var resp *http.Response
-	var result api.JottaFolder // Could be JottaFileDirList, but JottaFolder is close enough
 	err = f.pacer.Call(func() (bool, error) {
-		resp, err = f.srv.CallXML(ctx, &opts, nil, &result)
+		resp, err = f.srv.Call(ctx, &opts)
+		if err != nil {
+			return shouldRetry(ctx, resp, err)
+		}
+
+		// raw path is /username/device/mountpoint/path
+		// liststream paths are /mountpoint/path
+		trimPrefix := f.filePathRaw("")
+		trimPrefix = trimPrefix[strings.Index(trimPrefix, "/")+1:]
+		trimPrefix = trimPrefix[strings.Index(trimPrefix, "/"):]
```

Comment:
Instead of building the root path using filePathRaw and trimming stuff back of it would probably be cleaner to just build the path you need here.

Reply:
I agree. Will do that.